tools/SetupFlow/DevHome.SetupFlow/ViewModels/SetupFlowViewModel.cs

```diff
@@ -123,30 +126,52 @@ public async Task StartFileActivationFlowAsync(StorageFile file)
         await _mainPageViewModel.StartConfigurationFileAsync(file);
     }
 
-    public void StartCreationFlowAsync(string originPage)
+    public void StartCreationFlow(string originPage)
     {
         Orchestrator.FlowPages = [_mainPageViewModel];
 
         // This method is only called when the user clicks a button that redirects them to 'Create Environment' flow in the setup flow.
         _mainPageViewModel.StartCreateEnvironmentWithTelemetry(string.Empty, _creationFlowNavigationParameter, originPage);
     }
 
+    public void StartSetupFlow(string originPage, ComputeSystemReviewItem item)
+    {
+        Orchestrator.FlowPages = [_mainPageViewModel];
+
+        // This method is only called when the user clicks a button that redirects them to 'Setup' flow in the Environments page.
+        _mainPageViewModel.StartSetupForTargetEnvironmentWithTelemetry(string.Empty, _configurationFlowNavigationParameter, originPage, item);
+    }
+
     public void OnNavigatedTo(NavigationEventArgs args)
     {
         // The setup flow isn't setup to support using the navigation service to navigate to specific

```

Comment:
```suggestion
        // The setup flow isn't set up to support using the navigation service to navigate to specific
```
Once you start seeing it you can't stop 🙂